pytorch_lightning/metrics/classification/accuracy.py

```diff
@@ -16,33 +16,52 @@
 import torch
 
 from pytorch_lightning.metrics.metric import Metric
-from pytorch_lightning.metrics.utils import _input_format_classification
+from pytorch_lightning.metrics.functional.accuracy import _accuracy_update, _accuracy_compute
 
 
 class Accuracy(Metric):
     r"""
     Computes `Accuracy <https://en.wikipedia.org/wiki/Accuracy_and_precision>`_:
 
-    .. math:: \text{Accuracy} = \frac{1}{N}\sum_i^N 1(y_i = \hat{y_i})
+    .. math::
+        \text{Accuracy} = \frac{1}{N}\sum_i^N 1(y_i = \hat{y}_i)
 
     Where :math:`y` is a tensor of target values, and :math:`\hat{y}` is a
-    tensor of predictions.  Works with binary, multiclass, and multilabel
-    data.  Accepts logits from a model output or integer class values in
-    prediction.  Works with multi-dimensional preds and target.
+    tensor of predictions.
 
-    Forward accepts
+    For multi-class and multi-dimensional multi-class data with probability predictions, the
+    parameter ``top_k`` generalizes this metric to a Top-K accuracy metric: for each sample the
+    top-K highest probability items are considered to find the correct label.
 
-    - ``preds`` (float or long tensor): ``(N, ...)`` or ``(N, C, ...)`` where C is the number of classes
-    - ``target`` (long tensor): ``(N, ...)``
+    This metric generalizes to subset accuracy for multilabel data: for the sample to be counted as
+    correct, all labels in that sample have to be correctly predicted. Consider using :class:`~pytorch_lightning.metrics.classification.HammingLoss`
+    is this is not what you want. In a multi-dimensional multi-class case, the `mdmc_accuracy` parameters
+    gives you a choice between computing the subset accuracy or counting each sample on the extra
+    axis separately.
 
-    If preds and target are the same shape and preds is a float tensor, we use the ``self.threshold`` argument.
-    This is the case for binary and multi-label logits.
-
-    If preds has an extra dimension as in the case of multi-class scores we perform an argmax on ``dim=1``.
+    Accepts all input types listed in :ref:`metrics:Input types`.
 
     Args:
         threshold:
-            Threshold value for binary or multi-label logits. default: 0.5
+            Threshold probability value for transforming probability predictions to binary
+            `(0,1)` predictions, in the case of binary or multi-label inputs. Default: `0.5`
+        top_k:
+            Number of highest probability predictions considered to find the correct label, relevant
+            only for (multi-dimensional) multi-class inputs with probability predictions. The
+            default value (``None``) will be interpreted as 1 for these inputs.
+
+            Should be left at default (``None``) for all other types of inputs.
+        mdmc_accuracy:
```

Comment:
I wonder if we can come up with another name for this parameter, that is a bit more descriptive?

Reply:
Hm, how about if we replace this parameter with one called `subset_accuracy`, and set it to `False` by default? 

This would then have an effect in the case of multi-label inputs (`=True` means the whole sample has to be correct, `=False` would count each label separately), and for mdmc inputs (`=True` equivalent to `subset` now and `=False` equivalent to `global` now)?

Reply:
Ok, actually I think this is the perfect solution - enables us to keep the default behavior of the old accuracy for all cases, while giving users the option to compute subset_accuracy for ml and mdmc inputs.

I went ahead and implemented it, but let me know if you have some objections/other ideas.